scheduler/test/cook/test/impersonation.clj

```diff
@@ -0,0 +1,115 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns cook.test.impersonation
+  (:use clojure.test)
+  (:require [cook.authorization :as auth]
+            [cook.impersonation :as imp]
+            [cook.mesos.api :as api]))
+
+(def test-job-owner "the-job-owner")
+(def test-job {:owner test-job-owner :item :job})
+
+(def impersonator-user "the-impersonator")
+(def impersonator-job {:owner impersonator-user :item :job})
+
+(def admin-user "admin")
+
+(def auth-settings
+  {:authorization-fn 'cook.authorization/configfile-admins-auth-open-gets
+   :admins #{"admin" "other-admin"}})
+
+(def is-authorized? (imp/impersonation-authorized-wrapper auth/is-authorized? auth-settings))
+
+(deftest impersonated-is-authorized
+  (testing "that is-authorized? function performs correctly with impersonated users"
```

Comment:
I personally like the "that" in front, but are we doing that elsewhere? Might be better to be consistent (or change it everywhere)